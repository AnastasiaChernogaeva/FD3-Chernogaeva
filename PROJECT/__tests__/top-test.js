import React from 'react';
import renderer from 'react-test-renderer';
 
import Top from '../components/top';


test('работа Top', () => {
 
  // создаём тестовую версию компонента
  const component = renderer.create(
    <Top />
  );
 
  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
 
  // найдём в вёрстке компонента сами кнопки
 
  const buttonEleme1 = component.root.find( el =>(el.props.id=="FirstT")); 
  const buttonEleme2 = component.root.find( el =>(el.props.id=="SecT")); 
  const buttonEleme3 = component.root.find( el =>(el.props.id=="ThT")); 
  
  // const buttonEleme1 = component.root.find( el =>(el.props.testPrT1=="buttOn")); 
  // const buttonEleme2 = component.root.find( el =>(el.props.testPrT2=="buttOn")); 
  // const buttonEleme3 = component.root.find( el =>(el.props.testPrT3=="buttOn")); 

  const buttonElems=[buttonEleme1, buttonEleme2, buttonEleme3,];
  // находим все кнопки
 
  buttonElems.forEach(elem=>{elem.props.onClick(); // по очереди нажимаем на кнопки и делаем снимк(так два раза)
  
  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  // и "нажмём" на неё
 
  elem.props.onClick();
  
  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
 
})
})  