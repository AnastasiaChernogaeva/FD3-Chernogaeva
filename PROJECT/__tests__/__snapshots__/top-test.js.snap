// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Top 1`] = `
<div
  className="Top"
>
  <div
    className="TopNameAndSearch"
  >
    <p
      className="ShopName"
    />
    <p
      className="ShopNameBackGround"
    />
    <div
      className="Text"
    >
      <input
        className="SearchEngine"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <button
        className="button_search"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
          search
        </span>
      </button>
    </div>
  </div>
  <br />
  <div
    className="JustLikeThat"
  >
    <div
      className="account"
    >
      <p>
        <span
          className="material-icons Light"
        >
          account_circle
        </span>
      </p>
      <p />
      <p />
      <input
        className="Here"
        onClick={[Function]}
        type="button"
        value="Выйти"
      />
    </div>
     
  </div>
  <div
    className="Butt"
  >
    <div
      className="Top_Buttons First"
    >
      <input
        id="FirstT"
        onClick={[Function]}
        type="button"
        value="Главная"
      />
      <input
        id="SecT"
        onClick={[Function]}
        type="button"
        value="Корзина"
      />
      <input
        id="ThT"
        onClick={[Function]}
        type="button"
        value="WishList"
      />
    </div>
    <div
      className="Top_Buttons Second"
    >
       
    </div>
    <div
      className="Top_Buttons Third"
    >
      <input
        onClick={[Function]}
        type="button"
        value="Mои заказы"
      />
       
    </div>
  </div>
</div>
`;

exports[`работа Top 2`] = `
<div
  className="Top"
>
  <div
    className="TopNameAndSearch"
  >
    <p
      className="ShopName"
    />
    <p
      className="ShopNameBackGround"
    />
    <div
      className="Text"
    >
      <input
        className="SearchEngine"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <button
        className="button_search"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
          search
        </span>
      </button>
    </div>
  </div>
  <br />
  <div
    className="JustLikeThat"
  >
    <div
      className="account"
    >
      <p>
        <span
          className="material-icons Light"
        >
          account_circle
        </span>
      </p>
      <p />
      <p />
      <input
        className="Here"
        onClick={[Function]}
        type="button"
        value="Выйти"
      />
    </div>
     
  </div>
  <div
    className="Butt"
  >
    <div
      className="Top_Buttons First"
    >
      <input
        id="FirstT"
        onClick={[Function]}
        type="button"
        value="Главная"
      />
      <input
        id="SecT"
        onClick={[Function]}
        type="button"
        value="Корзина"
      />
      <input
        id="ThT"
        onClick={[Function]}
        type="button"
        value="WishList"
      />
    </div>
    <div
      className="Top_Buttons Second"
    >
       
    </div>
    <div
      className="Top_Buttons Third"
    >
      <input
        onClick={[Function]}
        type="button"
        value="Mои заказы"
      />
       
    </div>
  </div>
</div>
`;

exports[`работа Top 3`] = `
<div
  className="Top"
>
  <div
    className="TopNameAndSearch"
  >
    <p
      className="ShopName"
    />
    <p
      className="ShopNameBackGround"
    />
    <div
      className="Text"
    >
      <input
        className="SearchEngine"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <button
        className="button_search"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
          search
        </span>
      </button>
    </div>
  </div>
  <br />
  <div
    className="JustLikeThat"
  >
    <div
      className="account"
    >
      <p>
        <span
          className="material-icons Light"
        >
          account_circle
        </span>
      </p>
      <p />
      <p />
      <input
        className="Here"
        onClick={[Function]}
        type="button"
        value="Выйти"
      />
    </div>
     
  </div>
  <div
    className="Butt"
  >
    <div
      className="Top_Buttons First"
    >
      <input
        id="FirstT"
        onClick={[Function]}
        type="button"
        value="Главная"
      />
      <input
        id="SecT"
        onClick={[Function]}
        type="button"
        value="Корзина"
      />
      <input
        id="ThT"
        onClick={[Function]}
        type="button"
        value="WishList"
      />
    </div>
    <div
      className="Top_Buttons Second"
    >
       
    </div>
    <div
      className="Top_Buttons Third"
    >
      <input
        onClick={[Function]}
        type="button"
        value="Mои заказы"
      />
       
    </div>
  </div>
</div>
`;

exports[`работа Top 4`] = `
<div
  className="Top"
>
  <div
    className="TopNameAndSearch"
  >
    <p
      className="ShopName"
    />
    <p
      className="ShopNameBackGround"
    />
    <div
      className="Text"
    >
      <input
        className="SearchEngine"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <button
        className="button_search"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
          search
        </span>
      </button>
    </div>
  </div>
  <br />
  <div
    className="JustLikeThat"
  >
    <div
      className="account"
    >
      <p>
        <span
          className="material-icons Light"
        >
          account_circle
        </span>
      </p>
      <p />
      <p />
      <input
        className="Here"
        onClick={[Function]}
        type="button"
        value="Выйти"
      />
    </div>
     
  </div>
  <div
    className="Butt"
  >
    <div
      className="Top_Buttons First"
    >
      <input
        id="FirstT"
        onClick={[Function]}
        type="button"
        value="Главная"
      />
      <input
        id="SecT"
        onClick={[Function]}
        type="button"
        value="Корзина"
      />
      <input
        id="ThT"
        onClick={[Function]}
        type="button"
        value="WishList"
      />
    </div>
    <div
      className="Top_Buttons Second"
    >
       
    </div>
    <div
      className="Top_Buttons Third"
    >
      <input
        onClick={[Function]}
        type="button"
        value="Mои заказы"
      />
       
    </div>
  </div>
</div>
`;

exports[`работа Top 5`] = `
<div
  className="Top"
>
  <div
    className="TopNameAndSearch"
  >
    <p
      className="ShopName"
    />
    <p
      className="ShopNameBackGround"
    />
    <div
      className="Text"
    >
      <input
        className="SearchEngine"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <button
        className="button_search"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
          search
        </span>
      </button>
    </div>
  </div>
  <br />
  <div
    className="JustLikeThat"
  >
    <div
      className="account"
    >
      <p>
        <span
          className="material-icons Light"
        >
          account_circle
        </span>
      </p>
      <p />
      <p />
      <input
        className="Here"
        onClick={[Function]}
        type="button"
        value="Выйти"
      />
    </div>
     
  </div>
  <div
    className="Butt"
  >
    <div
      className="Top_Buttons First"
    >
      <input
        id="FirstT"
        onClick={[Function]}
        type="button"
        value="Главная"
      />
      <input
        id="SecT"
        onClick={[Function]}
        type="button"
        value="Корзина"
      />
      <input
        id="ThT"
        onClick={[Function]}
        type="button"
        value="WishList"
      />
    </div>
    <div
      className="Top_Buttons Second"
    >
       
    </div>
    <div
      className="Top_Buttons Third"
    >
      <input
        onClick={[Function]}
        type="button"
        value="Mои заказы"
      />
       
    </div>
  </div>
</div>
`;

exports[`работа Top 6`] = `
<div
  className="Top"
>
  <div
    className="TopNameAndSearch"
  >
    <p
      className="ShopName"
    />
    <p
      className="ShopNameBackGround"
    />
    <div
      className="Text"
    >
      <input
        className="SearchEngine"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <button
        className="button_search"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
          search
        </span>
      </button>
    </div>
  </div>
  <br />
  <div
    className="JustLikeThat"
  >
    <div
      className="account"
    >
      <p>
        <span
          className="material-icons Light"
        >
          account_circle
        </span>
      </p>
      <p />
      <p />
      <input
        className="Here"
        onClick={[Function]}
        type="button"
        value="Выйти"
      />
    </div>
     
  </div>
  <div
    className="Butt"
  >
    <div
      className="Top_Buttons First"
    >
      <input
        id="FirstT"
        onClick={[Function]}
        type="button"
        value="Главная"
      />
      <input
        id="SecT"
        onClick={[Function]}
        type="button"
        value="Корзина"
      />
      <input
        id="ThT"
        onClick={[Function]}
        type="button"
        value="WishList"
      />
    </div>
    <div
      className="Top_Buttons Second"
    >
       
    </div>
    <div
      className="Top_Buttons Third"
    >
      <input
        onClick={[Function]}
        type="button"
        value="Mои заказы"
      />
       
    </div>
  </div>
</div>
`;

exports[`работа Top 7`] = `
<div
  className="Top"
>
  <div
    className="TopNameAndSearch"
  >
    <p
      className="ShopName"
    />
    <p
      className="ShopNameBackGround"
    />
    <div
      className="Text"
    >
      <input
        className="SearchEngine"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <button
        className="button_search"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
          search
        </span>
      </button>
    </div>
  </div>
  <br />
  <div
    className="JustLikeThat"
  >
    <div
      className="account"
    >
      <p>
        <span
          className="material-icons Light"
        >
          account_circle
        </span>
      </p>
      <p />
      <p />
      <input
        className="Here"
        onClick={[Function]}
        type="button"
        value="Выйти"
      />
    </div>
     
  </div>
  <div
    className="Butt"
  >
    <div
      className="Top_Buttons First"
    >
      <input
        id="FirstT"
        onClick={[Function]}
        type="button"
        value="Главная"
      />
      <input
        id="SecT"
        onClick={[Function]}
        type="button"
        value="Корзина"
      />
      <input
        id="ThT"
        onClick={[Function]}
        type="button"
        value="WishList"
      />
    </div>
    <div
      className="Top_Buttons Second"
    >
       
    </div>
    <div
      className="Top_Buttons Third"
    >
      <input
        onClick={[Function]}
        type="button"
        value="Mои заказы"
      />
       
    </div>
  </div>
</div>
`;
